presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -120,6 +120,14 @@ public SqlQueryManager(ClusterMemoryManager memoryManager, QueryMonitor queryMon
     public void start()
     {
         queryTracker.start();
+        queryManagementExecutor.scheduleWithFixedDelay(() -> {
+            try {
+                checkForMemoryLeaks();
```

Comment:
Consider embedding in the below timer callback that exists below?

Reply:
Can you help me understand why we pulled this out into a separate independent check that's scheduled separately outside of `enforceMemoryLimits`?

Reply:
I added some comments in the code. PTAL